### PR TITLE
[ci] move macos x86_64 build to civ2

### PR DESCRIPTION
**Pull request by can-anyscale**
Move macos x86_64  build to civ2. Remove macos from civ1. Just copy the code as it is. Also fix test wheels logic.

Test:
- CI

### CODE REVIEW

File: ci/build/test-wheels.sh [hunk 1]
```diff
@@ -139,7 +139,7 @@ elif [[ "$platform" == "macosx" ]]; then
     "$PIP_CMD" install -q "$PYTHON_WHEEL"
 
     # Install the dependencies to run the tests.
-    "$PIP_CMD" install -q aiohttp aiosignal frozenlist 'pytest==7.0.1' requests proxy.py
+    "$PIP_CMD" install -q aiohttp aiosignal numpy frozenlist 'pytest==7.0.1' requests proxy.py
```
Comment (can-anyscale) [inline ci/build/test-wheels.sh:142]: wheel_test is failing on mac due to this missing dependency, fix it while i'm here

File: ci/ray_ci/macos/macos_ci_build.sh [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  ./ci/ci.sh test_wheels
+  # Build jars
+  bash ./java/build-jar-multiplatform.sh darwin
+  # Upload the wheels and jars
+  # We don't want to push on PRs, in fact, the copy_files will fail because unauthenticated.
+  if [ "$BUILDKITE_PULL_REQUEST" != "false" ]; then exit 0; fi
+  pip install -q docker aws_requests_auth boto3
+  # Upload to branch directory.
+  python .buildkite/copy_files.py --destination branch_wheels --path ./.whl
+  python .buildkite/copy_files.py --destination branch_jars --path ./.jar/darwin
```
Comment (aslonnie) [inline ci/ray_ci/macos/macos_ci_build.sh:46]: if the artifacts work, I would just remove these lines..

File: ci/ray_ci/macos/macos_ci_build.sh [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  . ./ci/ci.sh init && source ~/.zshenv
+  source ~/.zshrc
+  ./ci/ci.sh build
+  # Test wheels
+  ./ci/ci.sh test_wheels
+  # Build jars
+  bash ./java/build-jar-multiplatform.sh darwin
+  # Upload the wheels and jars
+  # We don't want to push on PRs, in fact, the copy_files will fail because unauthenticated.
+  if [ "$BUILDKITE_PULL_REQUEST" != "false" ]; then exit 0; fi
```
Comment (aslonnie) [inline ci/ray_ci/macos/macos_ci_build.sh:42]: check if it is on postmerge pipeline here? I do not want it to depend on if it is pull request triggered.

File: ci/ray_ci/macos/macos_ci_build.sh [hunk 4]
```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+
+set -ex
+
+export CI="true"
+export PYTHON="3.8"
```
Comment (aslonnie) [inline ci/ray_ci/macos/macos_ci_build.sh:6]: should this be 3.9 now?

File: ci/build/test-wheels.sh (hunk 1 above)
Comment (aslonnie) [reply ci/build/test-wheels.sh:142]: maybe pick this line in a separate PR?

File: ci/ray_ci/macos/macos_ci_build.sh (hunk 4 above)
Comment (can-anyscale) [reply ci/ray_ci/macos/macos_ci_build.sh:6]: some osx failed in 3.9 so we reverted it back to 3.8 for macos

File: ci/ray_ci/macos/macos_ci_build.sh [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  # Upload the wheels and jars
+  # We don't want to push on PRs, in fact, the copy_files will fail because unauthenticated.
+  if [ "$BUILDKITE_PULL_REQUEST" != "false" ]; then exit 0; fi
+  pip install -q docker aws_requests_auth boto3
+  # Upload to branch directory.
+  python .buildkite/copy_files.py --destination branch_wheels --path ./.whl
+  python .buildkite/copy_files.py --destination branch_jars --path ./.jar/darwin
+  # Upload to latest directory.
+  if [ "$BUILDKITE_BRANCH" = "master" ]; then python .buildkite/copy_files.py --destination wheels --path ./.whl; fi
+  if [ "$BUILDKITE_BRANCH" = "master" ]; then python .buildkite/copy_files.py --destination jars --path ./.jar/darwin; fi
```
Comment (aslonnie) [inline ci/ray_ci/macos/macos_ci_build.sh:49]: release branches should do the same? no?

Comment (aslonnie) [reply ci/ray_ci/macos/macos_ci_build.sh:49]: or maybe this is uploading to the "nightly" thingy.

**Review by aslonnie — APPROVED** [review]
approving since it does not seem to make things worse.

the script obviously have things that we can optimize and fine tune.

File: ci/build/test-wheels.sh (hunk 1 above)
Comment (can-anyscale) [reply ci/build/test-wheels.sh:142]: it takes 3.5 hour for the mac build job to finish and green and the change is also small so let's merge together

**Comment by can-anyscale** [comment]
yeah let's separate the tech-debt clean up from this

**Pull request merged**